### ISSUE TITLE
Some menus don't respect the theme applied

### ISSUE THREAD

**Issue by boblaspugna** [WillPower3309/swayfx#56]
**Swayfx Version:** 
0.1-1 from AUR

**Description of the issue:**
Opening some gtk apps, both gtk4 and gtk3, i.e.: Nautilus, Evince, Firefox....., some menus appear with a different color than the theme applied.

This doesn't happen with the default Sway.

I'm using adw-gtk3 theme for gtk3 app with custom colors in:

- gtk3 apps: ~/.config/gtk-3.0/gtk.css;
- gtk4 apps: ~/.config/gtk-4.0/gtk.css;

Here some examples:

**Nautilus with Swayfx**
![image](https://user-images.githubusercontent.com/25135325/202389910-2f75db61-97f9-4332-929f-aab8bcf4645a.png)

**Nautilus with Sway**
![image](https://user-images.githubusercontent.com/25135325/202390020-601dc62d-16f0-4111-a146-df2470f1bfbc.png)

**Firefox with Swayfx**
![image](https://user-images.githubusercontent.com/25135325/202390132-dd47f9f9-d13e-4b1e-8993-c7653c430daf.png)

**Comment by ErikReider** [comment]
Are you experiencing this issue with sway 1.7? My initial guess would be that it's an upstream issue that is solved in the sway-git package

**Comment by boblaspugna** [comment]
With Sway 1:1.7-10, from the normal repo of Arch, everything is fine, as shown in the second screenshot I attached.

**Comment by ErikReider** [comment]
So it's only the background color right? How did you change the default color to yellow?

**Comment by boblaspugna** [comment]
This is my ~/.config/gtk-4.0/gtk.css and ~/.config/gtk-3.0/gtk.css

``````/* solarized light colours */
@define-color window_bg_color #eee8d5;
@define-color window_fg_color #586e75;

@define-color accent_color #dbd6c4;
@define-color accent_bg_color #dbd6c4;
@define-color accent_fg_color #073642;

@define-color headerbar_bg_color #f5efde;
@define-color headerbar_fg_color #586e75;
@define-color headerbar_backdrop_color #fdf6e3;

@define-color popover_bg_color #fdf6e3;
@define-color popover_fg_color #073642;

@define-color dialog_bg_color @popover_bg_color;
@define-color dialog_fg_color @popover_fg_color;

@define-color view_bg_color #fdf6e3;
@define-color view_fg_color #073642;

@define-color card_bg_color #eee8d5;
@define-color card_fg_color #073642;

@define-color warning_bg_color #CB4B16;
@define-color warning_fg_color #eee8d5;
@define-color warning_color #CB4B16;
@define-color error_bg_color #DC322F;
@define-color error_fg_color #eee8d5;
@define-color error_color #DC322F;
@define-color success_bg_color #859900;
@define-color success_fg_color #eee8d5;
@define-color success_color #859900;
@define-color destructive_bg_color #D33682;
@define-color destructive_fg_color #eee8d5;
@define-color destructive_color #D33682;

**Comment by boblaspugna** [comment]
The issue appears also with non-gtk apps, like Telegram:

**Telegram with Swayfx**
![image](https://user-images.githubusercontent.com/25135325/202490076-08fc3c9e-1455-4098-815f-04aade5d7fdc.png)

**Telegram with Sway**
![image](https://user-images.githubusercontent.com/25135325/202490216-16ebd85f-3047-4c5c-9d35-7bbaa251a074.png)

In this case, it seems everything is using a grayscale.

**Comment by WillPower3309** [comment]
Hmm looks like it's related to our saturation feature. Will investigate

**Comment by WillPower3309** [comment]
Believe I've found the issue, will fix this evening

**Comment by ErikReider** [comment]
> Hmm looks like it's related to our saturation feature. Will investigate

Oh yeah! Now when you mention it, looks like the first image isn't saturated for some reason. Strange, the default should always be 1.0 😵‍💫

**Comment by WillPower3309** [comment]
> > Hmm looks like it's related to our saturation feature. Will investigate
> 
> Oh yeah! Now when you mention it, looks like the first image isn't saturated for some reason. Strange, the default should always be 1.0 😵‍💫

My guess is it was due to missingsetting saturation in https://github.com/WillPower3309/swayfx/blob/master/sway/desktop/render.c#L373-L384 so it initialized a value of 0. Easy fix!

**Comment by moilong** [comment]
Can confirm this because my current build 0.1 showing the same behaviour but previous one that is built before saturation feature added didn"t.

**Comment by WillPower3309** [comment]
Fixed, will release swayfx 0.1.1 to include the fix. You can install via the swayfx-git AUR package, or wait a day for 0.1.1 to drop
